packages/main/src/RPA/Excel/Files.py

```diff
@@ -1206,17 +1216,41 @@ def append_worksheet(self, name=None, content=None, header=False, start=None):
         if header and is_empty:
             sheet.append(columns)
 
-        for row in content:
-            values = [""] * len(columns)
-            for column, value in row.items():
-                try:
-                    index = columns.index(column)
-                    values[index] = value
-                except ValueError:
-                    pass
-            sheet.append(values)
+        if formatting_as_empty:
+            self._append_on_first_empty_based_on_values(content, sheet)
+        else:
+            for row in content:
+                values = [""] * len(columns)
+                for column, value in row.items():
+                    try:
+                        index = columns.index(column)
+                        values[index] = value
+                    except ValueError:
+                        pass
+                sheet.append(values)
+
+        self.active = sheet_name
+
+    def _append_on_first_empty_based_on_values(self, content, sheet):
+        first_empty = None
+        for row in sheet.rows:
+            values = []
+            for cell in row:
+                values.append(cell.value)
+            if all(value is None for value in values):
+                if not first_empty:
+                    first_empty = row[0].row
+            else:
+                first_empty = None
 
-        self.active = name
+        if first_empty:
+            for idx, src_row in enumerate(sheet.iter_rows(min_row=first_empty)):
```

Comment:
Is this producing the expected behaviour? (aka putting the `content` row(s) in the sheet starting with `first_empty`)

Shouldn't we `enumerate` the content instead and fill in cell by cell and row by row until we deplete the `content` (exactly like is done on the default logic [branch](https://github.com/robocorp/rpaframework/pull/599/files#diff-fd845ab66b96b3c8b93a98a0e5ff2466427deacdff2bc735a415c1f18b4cd4b3R1222-R1230))? Asking because if `first_empty` is let's say the last row in the sheet, we fill this one alone and rest of the rows in `content` won't get copied, right?

Reply:
🤔 Or maybe that's the reason behind this parameter, to fill in with values as long as there are enough empty rows already present in the sheet.

Reply:
Counter tested [example](https://github.com/robocorp/rpaframework/pull/599#issuecomment-1210733054) demonstrating the current behaviour.

Reply:
Fixed this part. Indeed we need to enumerate content :)